### ISSUE TITLE
Programmatically changing invalid value to empty/null value does not clear value in UI

### ISSUE THREAD

**Issue by drewharvey** [vaadin/flow-components#1696]
### Description
If a user inputs an invalid value (eg: "asdfasdf") and then you try to programmatically clear the value (set the value to null or `datePicker.getEmptyValue()`), the value in the text field does not change and remains as the original, invalid value (eg: "asdfasdf").

### Steps to reproduce
1. From the UI, input an invalid string into the date picker (eg: "asdf")
1. In the code, attempt to clear the field: `datePicker.setValue(datePicker.getEmptyValue())` or `datePicker.setValue(null)`

### Expected outcome
Value in the field should change to empty.

### Actual outcome
Value in the field does not change at all and still shows the original, invalid input.

### Example
![datepicker-clear](https://user-images.githubusercontent.com/2496955/74868626-69067d00-530b-11ea-9d03-197fe7273c46.gif)

```
   	 	DatePicker dp = new DatePicker();
   	 	dp.setValue(LocalDate.now());
   	 	Button btn = new Button("Clear", e -> dp.setValue(dp.getEmptyValue()));
   	 	add(dp, btn);
```

### Environment
Vaadin 14.1.17
Pure Java environment

**Comment by nittka** [comment]
see also #1778

**Comment by TatuLund** [comment]
Related to https://github.com/vaadin/flow-components/issues/2176

The value produced by DatePicker from invalid string will be null. Hence when you try to clear the value (aka set it to null) the server side state is unchanged and no value is pushed to client.

You could do brute force update via JavaScript:

` datePicker.getElement().executeJs("this.inputElement.value = ''")`

**Comment by nittka** [comment]
When implementing the clear functionality yourself, you can also set a valid value first before setting the value to null. The underlying problem still persists. The component's behaviour is not consistent:
The empty content (null) may be a valid value, but "abc" (also null) is not, and the Java API does not allow to distinguish these two cases. I repeat myslef, but I do not consider this minor or of low priority for such a fundamental component.

**Comment by TatuLund** [comment]
> and the Java API does not allow to distinguish these two cases.

That is correct and discussed e.g. on the ticket I linked. The client side validation feature of the fields is not enabled when using Flow. Thus field isInvalid is not "true" in this case.

**Comment by mailq** [comment]
Isn't it solved with Vaadin 24? I can't reproduce it anymore.

**Comment by web-padawan** [comment]
Fixed by https://github.com/vaadin/flow-components/pull/4413.
